addons/account/models/account_move.py

```diff
@@ -1189,8 +1190,14 @@ def _compute_payment_state(self):
                         new_pmt_state = 'partial'
                     elif invoice.matched_payment_ids.filtered(lambda p: not p.move_id and p.state == 'paid'):
                         new_pmt_state = invoice._get_invoice_in_payment_state()
+
+            if new_pmt_state == 'paid' and invoice.payment_state != 'paid':
+                invoice_paid += invoice
             invoice.payment_state = new_pmt_state
 
+        if invoice_paid:
+            invoice_paid._invoice_paid_hook()
```

Comment:
We don't want to call this on `NewId`, right?

Reply:
Why ? I feel like the decision to apply change should be on the "user" of `_invoice_paid_hook` and not generic.
If your use case should not apply to new_id, then you should juste filter them when overriding `_invoice_paid_hook`

Reply:
If you call it on NewId, won't it be sending mails on records that don't exist?